### PR TITLE
Chall12 jtrancos

### PR DESCRIPTION
**Pull request by jtrancoso**
This is my try for Chall12. Tried to work around the `return NULL` rule since I couldn't make it work.

First we check if the string is valid, to do so we look for pairs of `(` and `)`, if we have an odd count, we mark it as invalid and return `NULL`.

After that we do an `strdup` of the initial string. We look for a `(`, if we don't find any we return the string just as it is.

If we have found a `(` we save its position and then start looking for its closing pair. Once we've got them both we reverse everything that's inside it and we reverse again each individual bracket so they face the right direction after that. Then we continue iterating for each pair of brackets we find until we get to the centre of the string.

### CODE REVIEW

**Comment by jtrancoso** [comment]
RIP deadline, took to long writing the description.

File: chall12/jtrancos.c [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	int	end;
+	int	i;
+	int	q;
+
+	i = 0;
+	q = strlen(str);
+	if (check_error(str) == 0)
+		temp = strdup(str);
+	else
+		temp = strdup("NULL");
```
Comment (clafoutis42) [inline chall12/jtrancos.c:82]: What is this ? 😅You want to return `NULL` not `"NULL"`.

File: chall12/jtrancos.c [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	char	*temp;
+	int	start;
+	int	end;
+	int	i;
+	int	q;
+
+	i = 0;
+	q = strlen(str);
+	if (check_error(str) == 0)
+		temp = strdup(str);
```
Comment (clafoutis42) [inline chall12/jtrancos.c:80]: What if the dup fails and you get `NULL` ?

File: chall12/jtrancos.c [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	else
+		temp = strdup("NULL");
+	if (strchr(temp, '(') == NULL)
+		return (temp);
+	while (temp[i])
+	{
+		while (temp[i] != '(' && i < q)
+			i++;
+		start = i;
+		i++;
```
Comment (clafoutis42) [inline chall12/jtrancos.c:90]: Watch out, if you come to the end of the string on this subloop, you will increment `i` one more time and be passed the null terminator, which means that you will segfault on the main loop.

**Review by clafoutis42 — COMMENTED** [review]
Nice try ! Too bad that you took too long to write the description :(  
Watch out for buffer overflows !

File: chall12/jtrancos.c (hunk 2 above)
Comment (jtrancoso) [reply chall12/jtrancos.c:80]: Yeah, should have covered it with a `if strdup == NULL`

File: chall12/jtrancos.c (hunk 1 above)
Comment (jtrancoso) [reply chall12/jtrancos.c:82]: Don't know why it would not return NULL so I tried to be creative about it 😅

**Comment by jtrancoso** [comment]
Thanks for the feedback! Very knowledgeable as always :) Next time I will upload it and edit the description afterwards

**Pull request merged**